arbor/include/arbor/cable_cell.hpp

```diff
@@ -29,51 +29,153 @@ struct lid_range {
-struct cell_probe_total_ionic_current_density {
+// Voltage estimate [mV], reported against each cable in each control volume. Not interpolated.
+// Sample value type: `cell_sample_range`
+// Sample metadata type: `mcable_list`
+struct cell_probe_membrane_voltage_cell {};
+
+// Axial current estimate [nA] at `location`, interpolated.
+// Sample value type: `double`
+// Sample metadata type: `mlocation`
+struct cell_probe_axial_current {
     mlocation location;
 };
 
+// Total current density [A/m²] across membrane _excluding_ capacitive current at `location`.
+// Sample value type: `cell_sample_range`
+// Sample metadata type: `mlocation`
+struct cell_probe_total_ion_current_density {
+    mlocation location;
+};
+
+// Total ionic current [nA] across membrance _excluding_ capacitive current across components of the cell.
+// Sample value type: `cell_sample_range`
+// Sample metadata type: `mcable_list`
+struct cell_probe_total_ion_current_cell {};
+
+// Total membrance current [nA] across components of the cell.
+// Sample value type: `cell_sample_range`
+// Sample metadata type: `mcable_list`
+struct cell_probe_total_current_cell {};
+
 // Value of state variable `state` in density mechanism `mechanism` in CV at `location`.
+// Sample value type: `double`
+// Sample metadata type: `mlocation`
 struct cell_probe_density_state {
     mlocation location;
     std::string mechanism;
     std::string state;
 };
 
+// Value of state variable `state` in density mechanism `mechanism` across components of the cell.
+// Sample value type: `cell_sample_range`
+// Sample metadata type: `mcable_list`
+struct cell_probe_density_state_cell {
+    std::string mechanism;
+    std::string state;
+};
+
 // Value of state variable `key` in point mechanism `source` at target `target`.
+// Sample value type: `double`
+// Sample metadata type: `cell_probe_point_info`
 struct cell_probe_point_state {
     cell_lid_type target;
     std::string mechanism;
     std::string state;
 };
 
+// Value of state variable `key` in point mechanism `source` at every target with this mechanism.
+// Metadata has one entry of type cell_point_info for each matched (possibly coalesced) instance.
+// Sample value type: `cell_sample_range`
+// Sample metadata type: `std::vector<cell_point_info>`
+struct cell_probe_point_state_cell {
+    std::string mechanism;
+    std::string state;
+};
+
 // Current density [A/m²] across membrane attributed to the ion `source` at `location`.
+// Sample value type: `double`
+// Sample metadata type: `mlocation`
 struct cell_probe_ion_current_density {
     mlocation location;
     std::string ion;
 };
 
-// Ionic internal concentration [mmol/L] of ion `source` at `location`, possibly interpolated.
+// Total ionic current [nA] attributed to the ion `source` across components of the cell.
+// Sample value type: `cell_sample_range`
+// Sample metadata type: `mcable_list`
+struct cell_probe_ion_current_cell {
```

Comment:
The explosion of types makes the interface unwieldy (according to my personal tastes).

The challenge is to make documentation and an interface that users can quickly get to what they want (particularly the python wrapper) without spending much time having to read docs and look for previous examples.

I think a "stringy" interface might fit the bill. Do you think it should be implemented in C++, on top of these types, or only in the Python wrapper (I would still write the string->probe-type code in C++, but keep it in `arbor/python` to be called directly by python wrapper calls).

Reply:
For the C++ API, it's this, or an enum together with a bunch of enum-contigent variant values. The type interface is, I am certain, clearer and easier to follow than the alternative.

What we do in Python though is another story.